src/app/view/components/dot-add-persona-dialog/dot-add-persona-dialog.component.spec.ts

```diff
@@ -0,0 +1,165 @@
+import { By } from '@angular/platform-browser';
+import { DotHttpErrorManagerService } from '@services/dot-http-error-manager/dot-http-error-manager.service';
+import { LoginService, SiteService } from 'dotcms-js';
+import { LoginServiceMock } from '@tests/login-service.mock';
+import { mockDotPersona } from '@tests/dot-persona.mock';
+import { of as observableOf, throwError as observableThrowError } from 'rxjs';
+import { DotCreatePersonaFormComponent } from '@components/dot-add-persona-dialog/dot-create-persona-form/dot-create-persona-form.component';
+import { NgControl } from '@angular/forms';
+import { FileUploadModule } from 'primeng/primeng';
+import { SiteServiceMock } from '@tests/site-service.mock';
+import { mockResponseView } from '@tests/response-view.mock';
+
+@Component({
+    selector: 'dot-field-validation-message',
+    template: ''
+})
+class TestFieldValidationMessageComponent {
+    @Input() field: NgControl;
+    @Input() message: string;
+}
+
+@Component({
+    selector: 'dot-site-selector',
+    template: ''
+})
+class MockSiteSelectorComponent {
+    @Input() archive = false;
+    @Input() id = '';
+    @Input() live = true;
+    @Input() system = true;
+}
+
+describe('DotAddPersonaDialogComponent', () => {
+    let component: DotAddPersonaDialogComponent;
+    let fixture: ComponentFixture<DotAddPersonaDialogComponent>;
+    let dotDialog: DebugElement;
+    const messageServiceMock = new MockDotMessageService({
+        'modes.persona.add.persona': 'Add Persona',
+        'dot.common.dialog.accept': 'Accept',
+        'dot.common.dialog.reject': 'Cancel'
+    });
+
+    beforeEach(() => {
+        const siteServiceMock = new SiteServiceMock();
+
+        DOTTestBed.configureTestingModule({
+            declarations: [
+                DotAddPersonaDialogComponent,
+                DotCreatePersonaFormComponent,
+                TestFieldValidationMessageComponent,
+                MockSiteSelectorComponent
+            ],
+            imports: [BrowserAnimationsModule, DotDialogModule, FileUploadModule],
+            providers: [
+                DotWorkflowActionsFireService,
+                DotHttpErrorManagerService,
+                { provide: DotMessageService, useValue: messageServiceMock },
+                { provide: LoginService, useClass: LoginServiceMock },
+                { provide: SiteService, useValue: siteServiceMock }
+            ]
+        });
+
+        fixture = TestBed.createComponent(DotAddPersonaDialogComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+        dotDialog = fixture.debugElement.query(By.css('dot-dialog'));
+    });
+
+    it('should not be visible by default', () => {
+        expect(dotDialog).toBeNull();
+    });
+
+    describe('visible Dialog', () => {
+        beforeEach(() => {
+            component.visible = true;
+            fixture.detectChanges();
+            dotDialog = fixture.debugElement.query(By.css('dot-dialog'));
```

Comment:
dot-dialog is already in the dom, you don't have to query for it every time, you can just do it in the first beforeEach.

Reply:
here https://github.com/dotCMS/core-web/pull/1124/files/e262c800ad00a83a71cdf2dc5a4c800de8448bb8#diff-c2e8b519a03812cfdd256e23d0523d85R77 the dialog is null because visible is false, that is why I need run a query again.